frontend/src/components/Pagination/PaginationDesktop.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * Desktop variant for the Pagination component.
+ */
+
+import { useCallback } from 'react'
+import { BiChevronLeft, BiChevronRight } from 'react-icons/bi'
+import {
+  Box,
+  Button,
+  HStack,
+  IconButton,
+  Text,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+
+import { PAGINATION_THEME_KEY } from '~theme/components/Pagination'
+import { usePaginationRange } from '~hooks/usePaginationRange'
+
+import { PaginationProps } from './Pagination'
+
+// Separate constant to denote a separator in the pagination component.
+const SEPARATOR = '\u2026'
+
+interface DesktopPageButtonProps {
+  selectedPage: PaginationProps['currentPage']
+  page: number | typeof SEPARATOR
+  onClick: PaginationProps['onPageChange']
+  isDisabled: PaginationProps['isDisabled']
+}
+
+const DesktopPageButton = ({
+  selectedPage,
+  page,
+  onClick,
+  isDisabled,
+}: DesktopPageButtonProps) => {
+  const isSelected = page === selectedPage
+
+  const styles = useMultiStyleConfig(PAGINATION_THEME_KEY, { isSelected })
+
+  const handleClick = useCallback(() => {
+    if (page === SEPARATOR) return
+    onClick(page)
+  }, [onClick, page])
+
+  if (page === SEPARATOR) {
+    return <Text sx={styles.separator}>{page}</Text>
+  }
+
+  return (
+    <Button sx={styles.button} onClick={handleClick} isDisabled={isDisabled}>
+      {page}
+    </Button>
+  )
+}
+
+export const PaginationDesktop = ({
+  siblingCount = 1,
+  pageSize,
+  onPageChange,
+  totalCount,
+  currentPage,
+  isDisabled,
+}: PaginationProps): JSX.Element => {
+  const paginationRange = usePaginationRange<typeof SEPARATOR>({
+    totalCount,
+    pageSize,
+    currentPage,
+    siblingCount,
+    separator: SEPARATOR,
+  })
+
+  const styles = useMultiStyleConfig(PAGINATION_THEME_KEY, {})
+
+  const totalPageCount = Math.ceil(totalCount / pageSize)
+  const isDisableNextPage = isDisabled || currentPage >= totalPageCount
+  const isDisablePrevPage = isDisabled || currentPage <= 1
+
+  const handlePageBack = useCallback(() => {
+    if (isDisablePrevPage) return
+    onPageChange(currentPage - 1)
+  }, [currentPage, isDisablePrevPage, onPageChange])
+
+  const handlePageNext = useCallback(() => {
+    if (isDisableNextPage) return
+    onPageChange(currentPage + 1)
+  }, [currentPage, isDisableNextPage, onPageChange])
+
+  return (
+    <Box __css={styles.container}>
```

Comment:
could i ask why we aren't using the compound component pattern here? it would allow users to style the buttons/boundary separators, esp w/ exposing style props!

Reply:
As discussed on Slack, this component is not a layout component but its own component, don't think there is a real need to expose child components